### ISSUE TITLE
(Apple Silicon aka M1) import tensorflow killed python in terminal

### ISSUE THREAD

**Issue by J-H-Sung** [tensorflow/tensorflow#52138]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 11.6 (Big Sur)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version: 2.4, 2.5, 2.6
- Python version: 3.9.7
- Installed using virtualenv? pip? conda?: conda (Miniforge 3 ARM64), pip 
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: M1

**Describe the problem**
Hi, there. 
I have a problem with importing tensorflow related modules (tensorflow, keras, etc.). 
Others e.g., pandas, numpy, pillow, etc. worked without issue.
Obviously, I installed tensorflow supporting arm64 NOT Intel x86x64, and other modules such as PyTorch installed as ARM worked well without Rosseta.

After executing 'import tensorflow' in the terminal, it instantaneously kills python. (zsh: killed python)
I tried to change the python version to 3.8 and tensorflow version to 2.4 or 2.5, but results were the same. 
Reinstalling miniforge 3 ARM64 also did not work well.
Actually, importing tensorflow worked very well about just a week ago, but suddenly it didn't work :(

However, one strange thing is when I open the terminal with Rosetta, there is no issue for importing tensorflow and it uses GPU, too when I execute some toy example codes (MNIST).
Thanks!

**Provide the exact sequence of commands / steps that you executed before running into the problem**

1. tensorflow install command
conda install -c apple tensorflow-deps
conda install tensorflow
pip instal ltensorflow-metal

2. import tensorflow in terminal (M1 native)

Python 3.9.7 | packaged by conda-forge | (default, Sep 23 2021, 07:30:24) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
'>>> import tensorflow as tf
zsh: killed     python

3. import tensorflow in terminal with Rosetta

Python 3.9.7 | packaged by conda-forge | (default, Sep 23 2021, 07:30:24)
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.

'>>> import tensorflow as tf
Init Plugin
Init Graph Optimizer
Init Kernel

'>>> tf.test.is_gpu_available()
WARNING:tensorflow:From <stdin>:1: is_gpu_available (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.config.list_physical_devices('GPU')` instead.
Metal device set to: Apple M1

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB

2021-09-26 12:25:10.098855: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:305] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2021-09-26 12:25:10.099073: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:271] Created TensorFlow device (/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: METAL, pci bus id: <undefined>)
True

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**(tmp) jh@JHs-M1-MacBookAir ~ % conda install -c apple tensorflow-deps**
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

Package Plan

  environment location: /opt/homebrew/Caskroom/miniforge/base/envs/tmp

  added / updated specs:
    - tensorflow-deps

The following NEW packages will be INSTALLED:

  absl-py            conda-forge/noarch::absl-py-0.10.0-pyhd8ed1ab_1
  aiohttp            conda-forge/osx-arm64::aiohttp-3.7.4.post0-py39h5161555_0
  astunparse         conda-forge/noarch::astunparse-1.6.3-pyhd8ed1ab_0
  async-timeout      conda-forge/noarch::async-timeout-3.0.1-py_1000
  attrs              conda-forge/noarch::attrs-21.2.0-pyhd8ed1ab_0
  blinker            conda-forge/noarch::blinker-1.4-py_1
  brotlipy           conda-forge/osx-arm64::brotlipy-0.7.0-py39h5161555_1001
  c-ares             conda-forge/osx-arm64::c-ares-1.17.2-h3422bc3_0
  ca-certificates    conda-forge/osx-arm64::ca-certificates-2021.5.30-h4653dfc_0
  cached-property    conda-forge/noarch::cached-property-1.5.2-hd8ed1ab_1
  cached_property    conda-forge/noarch::cached_property-1.5.2-pyha770c72_1
  cachetools         conda-forge/noarch::cachetools-4.2.2-pyhd8ed1ab_0
  certifi            conda-forge/osx-arm64::certifi-2021.5.30-py39h2804cbe_0
  cffi               conda-forge/osx-arm64::cffi-1.14.6-py39h52b1de0_1
  chardet            conda-forge/osx-arm64::chardet-4.0.0-py39h2804cbe_1
  charset-normalizer conda-forge/noarch::charset-normalizer-2.0.0-pyhd8ed1ab_0
  click              conda-forge/osx-arm64::click-8.0.1-py39h2804cbe_0
  cryptography       conda-forge/osx-arm64::cryptography-3.4.7-py39h73257c9_0
  dataclasses        conda-forge/noarch::dataclasses-0.8-pyhc8e2a94_3
  flatbuffers        conda-forge/osx-arm64::flatbuffers-2.0.0-hbdafb3b_0
  gast               conda-forge/noarch::gast-0.4.0-pyh9f0ad1d_0
  google-auth        conda-forge/noarch::google-auth-1.35.0-pyh6c4a22f_0
  google-auth-oauth~ conda-forge/noarch::google-auth-oauthlib-0.4.6-pyhd8ed1ab_0
  google-pasta       conda-forge/noarch::google-pasta-0.2.0-pyh8c360ce_0
  grpcio             conda-forge/osx-arm64::grpcio-1.38.1-py39h9e1b6db_0
  h5py               conda-forge/osx-arm64::h5py-3.1.0-nompi_py39h99babb8_100
  hdf5               conda-forge/osx-arm64::hdf5-1.10.6-nompi_h0fc092c_1114
  idna               conda-forge/noarch::idna-3.1-pyhd3deb0d_0
  importlib-metadata conda-forge/osx-arm64::importlib-metadata-4.8.1-py39h2804cbe_0
  keras              conda-forge/noarch::keras-2.6.0-pyhd8ed1ab_0
  keras-preprocessi~ conda-forge/noarch::keras-preprocessing-1.1.2-pyhd8ed1ab_0
  krb5               conda-forge/osx-arm64::krb5-1.19.2-hd92b7a7_0
  libblas            conda-forge/osx-arm64::libblas-3.9.0-11_osxarm64_openblas
  libcblas           conda-forge/osx-arm64::libcblas-3.9.0-11_osxarm64_openblas
  libclang           conda-forge/osx-arm64::libclang-11.1.0-default_h0fdd720_1
  libcurl            conda-forge/osx-arm64::libcurl-7.79.1-h8fe1914_0
  libcxx             conda-forge/osx-arm64::libcxx-12.0.1-h168391b_0
  libedit            conda-forge/osx-arm64::libedit-3.1.20191231-hc8eb9b7_2
  libev              conda-forge/osx-arm64::libev-4.33-h642e427_1
  libffi             conda-forge/osx-arm64::libffi-3.4.2-hbdafb3b_4
  libgfortran        conda-forge/osx-arm64::libgfortran-5.0.0.dev0-11_0_1_hf114ba7_23
  libgfortran5       conda-forge/osx-arm64::libgfortran5-11.0.1.dev0-hf114ba7_23
  liblapack          conda-forge/osx-arm64::liblapack-3.9.0-11_osxarm64_openblas
  libllvm11          conda-forge/osx-arm64::libllvm11-11.1.0-h93073aa_2
  libnghttp2         conda-forge/osx-arm64::libnghttp2-1.43.0-hf3018f0_0
  libopenblas        conda-forge/osx-arm64::libopenblas-0.3.17-openmp_h5dd58f0_1
  libprotobuf        conda-forge/osx-arm64::libprotobuf-3.18.0-hccf11d3_0
  libssh2            conda-forge/osx-arm64::libssh2-1.10.0-hb80f160_1
  llvm-openmp        conda-forge/osx-arm64::llvm-openmp-12.0.1-hf3c4609_1
  markdown           conda-forge/noarch::markdown-3.3.4-pyhd8ed1ab_0
  multidict          conda-forge/osx-arm64::multidict-5.1.0-py39h5161555_1
  ncurses            conda-forge/osx-arm64::ncurses-6.2-h9aa5885_4
  numpy              conda-forge/osx-arm64::numpy-1.19.5-py39h1f3b974_2
  oauthlib           conda-forge/noarch::oauthlib-3.1.1-pyhd8ed1ab_0
  openssl            conda-forge/osx-arm64::openssl-1.1.1l-h3422bc3_0
  opt_einsum         conda-forge/noarch::opt_einsum-3.3.0-pyhd8ed1ab_1
  pip                conda-forge/noarch::pip-21.2.4-pyhd8ed1ab_0
  protobuf           conda-forge/osx-arm64::protobuf-3.18.0-py39hfb83b0d_0
  pyasn1             conda-forge/noarch::pyasn1-0.4.8-py_0
  pyasn1-modules     conda-forge/noarch::pyasn1-modules-0.2.7-py_0
  pycparser          conda-forge/noarch::pycparser-2.20-pyh9f0ad1d_2
  pyjwt              conda-forge/noarch::pyjwt-2.1.0-pyhd8ed1ab_0
  pyopenssl          conda-forge/noarch::pyopenssl-20.0.1-pyhd8ed1ab_0
  pysocks            conda-forge/osx-arm64::pysocks-1.7.1-py39h2804cbe_3
  python             conda-forge/osx-arm64::python-3.9.7-h54d631c_2_cpython
  python_abi         conda-forge/osx-arm64::python_abi-3.9-2_cp39
  pyu2f              conda-forge/noarch::pyu2f-0.1.5-pyhd8ed1ab_0
  readline           conda-forge/osx-arm64::readline-8.1-hedafd6a_0
  requests           conda-forge/noarch::requests-2.26.0-pyhd8ed1ab_0
  requests-oauthlib  conda-forge/noarch::requests-oauthlib-1.3.0-pyh9f0ad1d_0
  rsa                conda-forge/noarch::rsa-4.7.2-pyh44b312d_0
  scipy              conda-forge/osx-arm64::scipy-1.7.0-py39h5060c3b_0
  setuptools         conda-forge/osx-arm64::setuptools-58.0.4-py39h2804cbe_2
  six                conda-forge/noarch::six-1.15.0-pyh9f0ad1d_0
  sqlite             conda-forge/osx-arm64::sqlite-3.36.0-h72a2b83_2
  tensorboard        conda-forge/noarch::tensorboard-2.6.0-pyhd8ed1ab_1
  tensorboard-data-~ conda-forge/osx-arm64::tensorboard-data-server-0.6.0-py39hfb8cd70_0
  tensorboard-plugi~ conda-forge/noarch::tensorboard-plugin-wit-1.8.0-pyh44b312d_0
  tensorflow-deps    apple/osx-arm64::tensorflow-deps-2.6.0-0
  termcolor          conda-forge/noarch::termcolor-1.1.0-py_2
  tk                 conda-forge/osx-arm64::tk-8.6.11-he1e0b03_1
  typing-extensions  conda-forge/noarch::typing-extensions-3.7.4.3-0
  typing_extensions  conda-forge/noarch::typing_extensions-3.7.4.3-py_0
  tzdata             conda-forge/noarch::tzdata-2021a-he74cb21_1
  urllib3            conda-forge/noarch::urllib3-1.26.7-pyhd8ed1ab_0
  werkzeug           conda-forge/noarch::werkzeug-2.0.1-pyhd8ed1ab_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  wrapt              conda-forge/osx-arm64::wrapt-1.12.1-py39h5161555_3
  xz                 conda-forge/osx-arm64::xz-5.2.5-h642e427_1
  yarl               conda-forge/osx-arm64::yarl-1.6.3-py39h5161555_2
  zipp               conda-forge/noarch::zipp-3.5.0-pyhd8ed1ab_0
  zlib               conda-forge/osx-arm64::zlib-1.2.11-h31e879b_1009

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

**(tmp) jh@JHs-M1-MacBookAir ~ % conda install tensorflow**
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

Package Plan

  environment location: /opt/homebrew/Caskroom/miniforge/base/envs/tmp

  added / updated specs:
    - tensorflow

The following NEW packages will be INSTALLED:

  abseil-cpp         conda-forge/osx-arm64::abseil-cpp-20210324.2-hbdafb3b_0
  astor              conda-forge/noarch::astor-0.8.1-pyh9f0ad1d_0
  giflib             conda-forge/osx-arm64::giflib-5.2.1-h27ca646_2
  grpc-cpp           conda-forge/osx-arm64::grpc-cpp-1.37.1-h538f867_3
  icu                conda-forge/osx-arm64::icu-68.1-h17758a7_0
  jpeg               conda-forge/osx-arm64::jpeg-9d-h27ca646_0
  libpng             conda-forge/osx-arm64::libpng-1.6.37-hf7e6567_2
  python-flatbuffers conda-forge/noarch::python-flatbuffers-1.12-pyhd8ed1ab_1
  re2                conda-forge/osx-arm64::re2-2021.08.01-hbdafb3b_0
  snappy             conda-forge/osx-arm64::snappy-1.1.8-hc88da5d_3
  tensorflow         conda-forge/osx-arm64::tensorflow-2.6.0-py39hdf13c20_0
  tensorflow-base    conda-forge/osx-arm64::tensorflow-base-2.6.0-py39h34a2d98_0
  tensorflow-estima~ conda-forge/osx-arm64::tensorflow-estimator-2.6.0-py39h4ec10df_0

The following packages will be DOWNGRADED:

  grpcio                              1.38.1-py39h9e1b6db_0 --> 1.37.1-py39h9e1b6db_0
  libprotobuf                             3.18.0-hccf11d3_0 --> 3.15.8-hccf11d3_1
  protobuf                            3.18.0-py39hfb83b0d_0 --> 3.15.8-py39hfb83b0d_0

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

**(tmp) jh@JHs-M1-MacBookAir ~ % pip install tensorflow-metal**
Collecting tensorflow-metal
  Using cached tensorflow_metal-0.1.2-cp39-cp39-macosx_11_0_arm64.whl (1.2 MB)
Requirement already satisfied: wheel~=0.35 in /opt/homebrew/Caskroom/miniforge/base/envs/tmp/lib/python3.9/site-packages (from tensorflow-metal) (0.35.1)
Requirement already satisfied: six~=1.15.0 in /opt/homebrew/Caskroom/miniforge/base/envs/tmp/lib/python3.9/site-packages (from tensorflow-metal) (1.15.0)
Installing collected packages: tensorflow-metal
Successfully installed tensorflow-metal-0.1.2

**Comment by J-H-Sung** [comment]
I resolved myself this issue by following guide from this link: https://github.com/ctrahey/m1-tensorflow-config, but I still wonder why tensorflow installed via above procedure killed python.

**Comment by SeungVictor** [comment]
Did you update bigsur 11.6 recently? I did and I bumped into the same issue, I solved  the problem by upgrading to monterey 12(beta). I think bigsur 11.6 was a problem

**Comment by J-H-Sung** [comment]
@SeungVictor Thanks for the reply. Yes, I updated may Mac OS to Big Sur 11.6 and Xcode to 13.0, recently. I did not test my tensorflow codes right after the update, so that maybe I would not checked the issue right away. I'll try to check whether the issue still occurs on the Monterey OS.

**Comment by J-H-Sung** [comment]
Update 2: My tensorflow codes worked well after conda packages update using conda update conda and conda using conda update --all. Maybe the issue was a temporal crash related conda packages.

**Python 3.9.7 | packaged by conda-forge | (default, Sep 23 2021, 07:30:24) to Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:24:02)** 

Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:24:02) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Init Plugin
Init Graph Optimizer
Init Kernel
>>> tf.__version__
'2.6.0'
>>> tf.config.list_physical_devices()
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
